tests/integration/backend/filestate/backend_azure_test.go

```diff
@@ -0,0 +1,85 @@
+package filestate
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func loginAndCreateStack(t *testing.T, cloudURL string) {
+	t.Helper()
+	out, err := exec.Command("pulumi", "login", cloudURL).CombinedOutput()
+	require.NoError(t, err, string(out))
+
+	t.Setenv("PULUMI_CONFIG_PASSPHRASE", "test")
+	out, err = exec.Command("pulumi", "stack", "init", "testing").CombinedOutput()
+	require.NoError(t, err, string(out))
+	defer func() {
+		out, err := exec.Command("pulumi", "stack", "rm", "--yes", "-s", "testing").CombinedOutput()
+		assert.NoError(t, err, string(out))
+	}()
+
+	out, err = exec.Command("pulumi", "stack", "select", "testing").CombinedOutput()
+	require.NoError(t, err, string(out))
+
+	out, err = exec.Command("pulumi", "stack", "ls").CombinedOutput()
+	assert.NoError(t, err, string(out))
+	assert.Contains(t, string(out), "testing*")
+}
+
+//nolint:paralleltest // this test sets the global login state
+func TestAzureLoginSasToken(t *testing.T) {
+	err := os.Chdir("project")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := os.Chdir("..")
+		require.NoError(t, err)
+	})
+	cloudURL := "azblob://pulumitesting?storage_account=pulumitesting"
+	assert.NotEmpty(t, os.Getenv("AZURE_STORAGE_SAS_TOKEN"),
```

Comment:
So that people can run the tests without having all cloud credentials these should probably be `t.Skip` calls if the required envvars aren't set. We do this with quite a few tests already that require pulumi Cloud credentials.

Reply:
Makes sense, I've done that now.